src/fstext/determinize-lattice-inl.h

```diff
@@ -547,6 +547,11 @@ template<class Weight, class IntType> class LatticeDeterminizer {
       return (state != other.state || string != other.string ||
               weight != other.weight);
     }
+    // This operator is only intended to support sorting in the function
```

Comment:
Can you please implement an operator < instead?  That's the operator the sorting looks for by default, and in that case you wouldn't have to provide std::greater<Element>() to the function.  Also, this way we'll get the same order that the old code would have produced, which doesn't really matter but which is maybe less surprising.